src/Symfony/Component/AccessToken/AccessTokenFetcher.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AccessToken;
+
+/**
+ * @author Pierre Rineau <pierre.rineau@processus.org>
+ */
+class AccessTokenFetcher
+{
+    public function __construct(
+        private readonly AccessTokenManagerInterface $accessTokenManager,
+        private readonly CredentialsInterface $credentials,
+    ) {
+    }
+
+    /**
+     * Create instance using given URI.
+     */
+    public static function createWithUri(AccessTokenManagerInterface $accessTokenManager, #[\SensitiveParameter] string $uri): self
+    {
+        return new self($accessTokenManager, $accessTokenManager->createCredentials($uri));
+    }
+
+    /**
+     * Create credentials interface with given URI.
+     */
+    public static function createCredentialsWithUri(AccessTokenManagerInterface $accessTokenManager, #[\SensitiveParameter] string $uri): CredentialsInterface
+    {
+        return $accessTokenManager->createCredentials($uri);
+    }
+
+    /**
+     * Get access token.
```

Comment:
can be removed

Reply:
Not that sure, people that will inject the `AccessTokenFetcher` probably will want to have some documentation. But I agree that in this state, the comment is useless, it probably should be more descriptive.